src/app/workflow/workflow.component.html

```diff
@@ -156,20 +150,23 @@ <h3>
       </tab>
       <tab id="toolsTab"  heading="Tools" (select)="setEntryTab('tools')">
         <div *ngIf="validVersions && !isStub()">
-          <app-tool-tab [selectedVersion]="selectedVersion"></app-tool-tab>
+          <app-tool-tab [selectedVersion]="selectedVersion" [canRead]="canRead" [canWrite]="canWrite" [isOwner]="isOwner"></app-tool-tab>
         </div>
         <div *ngIf="isStub()" class="alert alert-warning" role="alert">
           <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span> To see tools, please refresh the workflow.
         </div>
       </tab>
       <tab id="dagTab" heading="DAG" (select)="setEntryTab('dag')">
         <div *ngIf="validVersions && !isStub() && currentTab == 'dag'">
-          <app-dag [id]="workflow?.id" [validVersions]="validVersions" [defaultVersion]="defaultVersion" [selectedVersion]="selectedVersion"></app-dag>
+          <app-dag [id]="workflow?.id" [validVersions]="validVersions" [defaultVersion]="defaultVersion" [selectedVersion]="selectedVersion" [canRead]="canRead" [canWrite]="canWrite" [isOwner]="isOwner"></app-dag>
         </div>
         <div *ngIf="isStub()" class="alert alert-warning" role="alert">
           <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span> To see the DAG, please refresh the workflow.
         </div>
       </tab>
+      <tab *ngIf="!isPublic() && isHosted() && isUserOwner()" id="permissionsTab" heading="Permissions" (select)="setEntryTab('permissions')">
```

Comment:
`isUserOwner()` expects username as a parameter, so permissions tab never shows up.

Reply:
I don't know how this works for you making this change, but I saw you demoing it. :)